src/puppetlabs/puppetdb/scf/storage_utils.clj

```diff
@@ -35,6 +35,34 @@
 
 ;; FUNCTIONS
 
+(defn deregister-unused-drivers
+  "Given a database config or connection pool, deregister the driver
+   (hsql or postgres) that isn't being used, and return the initial driver
+   list for reregistration in case it's required."
+  [db-config]
+  (let [configured-driver (java.sql.DriverManager/getDriver
+                            (if-let [datasource (:datasource db-config)]
+                              (.getJdbcUrl datasource)
+                              (format "jdbc:%s:%s"
+                                      (:subprotocol db-config)
+                                      (:subname db-config))))
+        initial-drivers (enumeration-seq (java.sql.DriverManager/getDrivers))]
+    (doseq [driver initial-drivers
+            :when (not= (class driver) (class configured-driver))]
+      (java.sql.DriverManager/deregisterDriver driver))
+    initial-drivers))
+
+(defn register-drivers
```

Comment:
Similarly a `!` here might be useful to denote side-effects. Also should this return the list of all registered drivers?